components/aws/sagemaker/train/component.yaml

```diff
@@ -82,13 +82,22 @@ inputs:
   - name: traffic_encryption
     description: 'Encrypts all communications between ML compute instances in distributed training.'
     default: 'False'
+  - name: spot_instance
+    description: 'Use managed spot training.'
+    default: 'False'
+  - name: max_wait_time
+    description: 'The maximum time in seconds you are willing to wait for a managed spot training job to complete.'
+    default: '86400'
+  - name: checkpoint_config
+    description: 'Dictionary of information about the output location for managed spot training checkpoint data.'
+    default: '{}'
```

Comment:
JFYI: The recent SDK (0.1.29+) supports the `JsonObject` type: arguments passed to `type: JsonObject` inputs will be serialized as JSON.

Reply:
Thanks for the tip!

Reply:
Does this update support nicer/easier input in the UI for JSON fields?

Reply:
Not yet. For now it only works for constant lists passed to components. The auto-conversion does not work in UX or when passing arguments to pipeline during the submission.

Reply:
So this will only work for this case:
```python
aws_sagemaker_trainop(..., checkpoint_config={'a' = 1, 'b' = 1},...)
```

Previously this would have just resulted in some wrong string (produced by `str()`) being passed. Now it would be `json.dumps`-serialized string.

Reply:
This is a great addition. Is there any way to suppress the warning that `dsl-compile` throws when it infers the JsonObject type?

Reply:
> Is there any way to suppress the warning that `dsl-compile` throws when it infers the JsonObject type?

The warning is shown every time a non-string argument is passed to an untyped input.
The easiest way to make it go away is to add the type to the input. Then there would be no inference.

Reply:
> The easiest way to make it go away is to add the type to the input.
Through explicit cast?

Reply:
> Through explicit cast?

No. Just add the `type` attribute to the input annotation:

```yaml
  - name: checkpoint_config
    type: JsonObject
    description: 'Dictionary of information about the output location for managed spot training checkpoint data.'
    default: '{}'
```